src/test/integration/index.ts

```diff
@@ -12,28 +12,22 @@ export function run(): Promise<void> {
 
   const testsRoot = path.resolve(__dirname, '.');
 
-  return new Promise((c, e) => {
-    glob('./**/**.test.js', { cwd: testsRoot }, (err, files) => {
-      if (err) {
-        return e(err);
-      }
+  try {
+    const files = await glob('./**/**.test.js', { cwd: testsRoot });
 
-      // Add files to the test suite
-      files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
+    files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
 
-      try {
-        // Run the mocha test
-        mocha.run(failures => {
-          if (failures > 0) {
-            e(new Error(`${failures} tests failed.`));
-          } else {
-            c();
-          }
-        });
-      } catch (err) {
-        console.error(err);
-        e(err);
-      }
+    return new Promise((resolve, reject) => {
+      mocha.run(failures => {
+        if (failures > 0) {
+          reject(new Error(`${failures} tests failed.`));
+        } else {
+          resolve();
+        }
+      });
     });
-  });
+  } catch (err) {
+    console.error(err);
+    throw err; // Rethrow the error for the caller to handle
+  }
```

Comment:
This refactor was needed after upgrading Glob. It highlighted that the returned Promise was not handled.